### PR TITLE
validator submission

### PR DESCRIPTION
**Pull request by epintozzi**

### CODE REVIEW

**Comment by jcasimir** [comment]
Got it. I'd like to see normal automated tests.

File: validator.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'pry'
+
+class StringValidator
+
+  BRACKET_HASH= {
```
Comment (mikedao) [inline validator.rb:5]: Great use of a constant here.

File: validator.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "(" => ")",
+    "{" => "}",
+    "[" => "]"
+  }
+
+  def validate(string)
+    return false if string.length < 2
+    brackets = []
```
Comment (mikedao) [inline validator.rb:13]: Do you think that `each_char` would be more or less efficient then chars-ing and then mapping?

File: validator.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    string.each_char do |x|
+      brackets.push(x) if BRACKET_HASH.keys.include?(x)
+      brackets.pop if BRACKET_HASH.values.include?(x) && (BRACKET_HASH[brackets.last] == x)
+    end
+    puts brackets.empty?
+  end
+
+end
+
+v = StringValidator.new
```
Comment (mikedao) [inline validator.rb:23]: Like Jeff said, would loved to have seen tests.

**Review by mikedao — APPROVED** [review]

**Pull request closed without merging**